builtin/providers/azurerm/config.go

```diff
@@ -0,0 +1,214 @@
+	client.secGroupClient = sgc
+
+	src := network.NewSecurityRulesClient(c.SubscriptionID)
+	src.Authorizer = spt
+	client.secRuleClient = src
+
+	snc := network.NewSubnetsClient(c.SubscriptionID)
+	snc.Authorizer = spt
+	client.subnetClient = snc
+
+	vgcc := network.NewVirtualNetworkGatewayConnectionsClient(c.SubscriptionID)
+	vgcc.Authorizer = spt
+	client.vnetGatewayConnectionsClient = vgcc
+
+	vgc := network.NewVirtualNetworkGatewaysClient(c.SubscriptionID)
+	vgc.Authorizer = spt
+	client.vnetGatewayClient = vgc
+
+	vnc := network.NewVirtualNetworksClient(c.SubscriptionID)
+	vnc.Authorizer = spt
+	client.vnetClient = vnc
+
+	rgc := resources.NewGroupsClient(c.SubscriptionID)
+	rgc.Authorizer = spt
+	client.resourceGroupClient = rgc
+
+	tc := resources.NewTagsClient(c.SubscriptionID)
+	tc.Authorizer = spt
+	client.tagsClient = tc
+
+	jc := scheduler.NewJobsClient(c.SubscriptionID)
+	jc.Authorizer = spt
+	client.jobsClient = jc
+
+	jcc := scheduler.NewJobCollectionsClient(c.SubscriptionID)
+	jcc.Authorizer = spt
+	client.jobsCollectionsClient = jcc
+
+	ssc := storage.NewAccountsClient(c.SubscriptionID)
+	ssc.Authorizer = spt
+	client.storageServiceClient = ssc
+
+	suc := storage.NewUsageOperationsClient(c.SubscriptionID)
+	suc.Authorizer = spt
+	client.storageUsageClient = suc
+
+	return &client, nil
+}
+
+// armCredentialsProvided is a helper method which indicates whether or not the
+// credentials required for authenticating against the ARM APIs were provided.
+func (c *Config) armCredentialsProvided() bool {
+	return c.ArmCredentials != "" || c._armCredentialsProvided()
+}
+func (c *Config) _armCredentialsProvided() bool {
+	return !(c.SubscriptionID == "" || c.ClientID == "" || c.ClientSecret == "" || c.TenantID == "")
+}
+
+// readArmSettings is a helper method which; given the contents of the ARM
+// credentials file, loads all the data into the Config.
+func (c *Config) readArmSettings(contents string) error {
+	data := &armConfigData{}
+	err := json.Unmarshal([]byte(contents), data)
+
+	c.SubscriptionID = data.SubscriptionID
+	c.ClientID = data.ClientID
+	c.ClientSecret = data.ClientSecret
+	c.TenantID = data.TenantID
+
+	return err
+}
+
+// validateArmConfigFile is a helper function which verifies that
+// the provided ARM configuration file is valid.
+func validateArmConfigFile(v interface{}, _ string) (ws []string, es []error) {
+	value := v.(string)
+	if value == "" {
+		return nil, nil
+	}
+
+	pathOrContents, _, err := pathorcontents.Read(v.(string))
+	if err != nil {
+		es = append(es, fmt.Errorf("Error reading 'arm_config_file': %s", err))
+	}
+
+	data := armConfigData{}
+	err = json.Unmarshal([]byte(pathOrContents), &data)
+	if err != nil {
+		es = append(es, fmt.Errorf("Error unmarshalling the provided 'arm_config_file': %s", err))
+	}
+
+	return
+}
+
+// armConfigData is a private struct which represents the expected layout of
+// an ARM configuration file. It is used for unmarshalling purposes.
+type armConfigData struct {
```

Comment:
forgot to add a TODO: the config file's format is inspired by the go SDK guys'[examples](https://github.com/Azure/azure-sdk-for-go/blob/master/arm/examples/helpers/credentials_sample.json), so we may feel fully free to re-define or even disregard this format...